scripts/build_powerplants.py

```diff
@@ -253,16 +253,34 @@ def add_custom_powerplants(ppl, inputs, config):
         return add_ppls
 
 
-def replace_natural_gas_technology(df):
-    mapping = {"Steam Turbine": "CCGT", "Combustion Engine": "OCGT"}
-    tech = df.Technology.replace(mapping).fillna("CCGT")
-    return df.Technology.mask(df.Fueltype == "Natural Gas", tech)
-
-
-def replace_natural_gas_fueltype(df):
-    return df.Fueltype.mask(
-        (df.Technology == "OCGT") | (df.Technology == "CCGT"), "Natural Gas"
+def replace_natural_gas_technology(df: pd.DataFrame):
+    """
+    Maps and replaces gas technologies in the powerplants.csv onto model
+    compliant carriers.
+    """
+    mapping = {
+        # "Steam Turbine": "CCGT",
+        "Combustion Engine": "OCGT",
+        "NG": "CCGT",
+        "Ng": "CCGT",
+        "NG/FO": "OCGT",
+        "Ng/Fo": "OCGT",
+        "NG/D": "OCGT",
+        "LNG": "OCGT",
+        "CCGT/D": "CCGT",
+        "CCGT/FO": "CCGT",
+        "LCCGT": "CCGT",
+        "CCGT/Fo": "CCGT",
+    }
+    df["Technology"] = df["Technology"].replace(mapping).fillna("CCGT")
+    unique_tech_with_ng = df[df["Fueltype"] == "Natural Gas"]["Technology"].unique()
+    unknown_techs = np.setdiff1d(unique_tech_with_ng, ["CCGT", "OCGT"])
+    if len(unknown_techs) > 0:
+        df["Technology"] = df["Technology"].map({t: "CCGT" for t in unknown_techs})
```

Comment:
I agree with the proposal above: only the tecs related to gas shall be revised.
This line overwrites also lines where the fuel type is not gas.
The pandas.where may be used both here and below.

For example:
`df.Technology.where(df.Technology == "Natural Gas", df["Technology"].map({t: "CCGT" for t in unknown_techs}), inplace=True)`

Reply:
done.